bin/plugin/commands/build-plugins.js

```diff
@@ -0,0 +1,178 @@
+
+						fs.copySync( modulePath, buildModulePath, { overwrite: true } );
+					} catch ( copyError ) {
+						log(
+							formats.error(
+								`Error copying plugin file: "${ copyError }"`
+							)
+						);
+						continue;
+					}
+
+					// Update file content.
+					updatePluginHeader( {
+						key: moduleDir,
+						slug: pluginSlug,
+						version: pluginVersion,
+						pluginModulePath: modulePath,
+					} );
+
+					// Update text domain.
+					updateTextDomain( {
+						slug: pluginSlug,
+					} );
+				} catch ( error ) {
+					log(
+						formats.error(
+							`${ error }`
+						)
+					);
+				}
+			}
+		} catch ( jsonError ) {
+			log(
+				formats.error(
+					`Error parsing JSON string: "${ jsonError }"`
+				)
+			);
+		}
+	} );
+};
+
+/**
+ * Updates the `load.php` file header information.
+ *
+ * @param {[]Settings} settings Plugin settings.
+ */
+async function updatePluginHeader( settings ) {
+	const { key, version, slug, pluginModulePath } = settings;
+	const loadFile = path.join( '.', 'load.php' );
+	const loadFileContent = fs.readFileSync( loadFile, 'utf-8' );
+	const buildLoadFile = path.join( '.', 'build/' + slug + '/load.php' );
+	const buildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+
+	const regex = /\/\\*\\*[\s\S]+?(?=\*\/)/mi;
+	const pluginComment = loadFileContent.match( regex )?.[ 0 ];
+	const newPluginComment = buildLoadFileContent.match( regex )?.[ 0 ];
+
+	fs.writeFileSync( buildLoadFile, buildLoadFileContent.replace( newPluginComment, pluginComment ) );
+
+	let updatedBuildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+
+	const moduleFilePattern = path.join( pluginModulePath, 'load.php' );
+	const moduleFiles = path.resolve( '.', moduleFilePattern );
+	const { name, description } = await getModuleDataFromHeaders( moduleFiles );
+
+	// Get module header data.
+	const moduleData = {
+		pluginname: name,
+		plugindescription: description,
+		pluginuri: '/tree/trunk/modules/' + key,
+		pluginversion: version,
+		textdomain: slug,
+	};
+
+	// Map of module header => object property.
+	const headers = {
+		'Plugin Name': 'pluginname',
+		'Plugin URI': 'pluginuri',
+		Description: 'plugindescription',
+		Version: 'pluginversion',
+		'Text Domain': 'textdomain',
+	};
+
+	const headerRegex = new RegExp( `(?<header>${ Object.keys( headers ).join( '|' ) }): (?<value>(.*?)+)`, 'gmi' );
+	const matches = [ ...updatedBuildLoadFileContent.matchAll( headerRegex ) ];
+
+	matches.forEach( ( match ) => {
+		const { header, value } = match.groups;
+		const prop = headers[ header ];
+		let moduleDataValue = moduleData[ prop ];
+
+		if ( value && prop && moduleDataValue ) {
+			if ( prop === 'pluginuri' ) {
+				moduleDataValue = value + moduleDataValue;
+			}
+			fs.writeFileSync( buildLoadFile, updatedBuildLoadFileContent.replace( new RegExp( value ), moduleDataValue ) );
+			updatedBuildLoadFileContent = fs.readFileSync( buildLoadFile, 'utf-8' );
+		}
+	} );
```

Comment:
Rather than having all those individual regexes, let's create the single plugin header string and make only one replacement. This should be straightforward after you have implemented and are using the functions I outlined in https://github.com/WordPress/performance/pull/662#discussion_r1126861733.

Reply:
Thanks @felixarntz Can you please share an example of it or more details?

Reply:
@mukeshpanchal27 See my reply in https://github.com/WordPress/performance/pull/662#discussion_r1128161824, I would suggest we don't read the original Performance Lab plugin header, but compute the entire header string ourselves. This simplifies things drastically and you can get rid of the additional regexes in `updatePluginHeader()`.

Here's the flow of logic I would suggest for your `updatePluginHeader()` function:
1. Get the content of the `load.php` file within `settings.pluginPath` (you're already doing that).
2. Extract the module header from it (you're already doing that).
3. Get the module data from the module header (you're already doing that).
4. Create the plugin header from scratch, e.g. with a new variable where you put the entire multiline string inside, starting with `/**\n * Plugin Name: ...`.
    * You can use a JS template string to fill in all the variable data for `Plugin Name`, `Description`, `Version`, `Text Domain`.
    * The `Plugin URI` field can be hard-coded to `https://github.com/WordPress/performance/tree/trunk/modules/{originalModulePath}`. For this you may need to pass a new parameter to the function.
    * The `Requires at least`, `Requires PHP`, `Author`, `Author URI`, `License`, and `License URI` parameters can all be hard-coded, to the same values that are used in the Performance Lab plugin header.
    * Don't forget to include the `@package` note at the end of the comment.
5. Replace the entire module header from point 2. with the new plugin header from point 4. within the overall file content you got in point 1.
6. Update the file from point 1. with the new content from point 5.

All this should be simpler than the regular expressions etc that you are currently using in this function.

Reply:
Thanks @felixarntz This feedback really solve all the problems that i face.